### PR TITLE
fix ast ListImports() and ListExports() panic with nil pointer

### PR DESCRIPTION
**Pull request by loredanacirstea**
closes https://github.com/second-state/WasmEdge-go/issues/37

### CODE REVIEW

File: wasmedge/ast.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ func (self *AST) ListImports() []*ImportType {
 			C.WasmEdge_ASTModuleListImports(self._inner, &(cimptype[0]), ltypes)
 		}
 		for i, val := range cimptype {
+			imptype[i] = &ImportType{}
```
Comment (q82419) [inline wasmedge/ast.go:63]: I think move the whole for loop into the if statement is better?

```go
if uint(ltypes) > 0 {
	imptype = make([]*ImportType, uint(ltypes))
	cimptype = make([]*C.WasmEdge_ImportTypeContext, uint(ltypes))
	C.WasmEdge_ASTModuleListImports(self._inner, &(cimptype[0]), ltypes)
	for i, val := range cimptype {
		imptype[i]._inner = val
		imptype[i]._ast = self._inner
		imptype[i]._own = false
	}
	return imptype
} else {
	return nil
}
```

File: wasmedge/ast.go [hunk 2]
```diff
@@ -80,6 +81,7 @@ func (self *AST) ListExports() []*ExportType {
 			C.WasmEdge_ASTModuleListExports(self._inner, &(cexptype[0]), ltypes)
 		}
 		for i, val := range cexptype {
+			exptype[i] = &ExportType{}
```
Comment (q82419) [inline wasmedge/ast.go:84]: As above

File: wasmedge/ast.go (hunk 1 above)
Comment (loredanacirstea) [reply wasmedge/ast.go]: done in https://github.com/second-state/WasmEdge-go/pull/38/commits/38caaf64a4b7c60cce3c825074232e124b9141a0

**Comment by loredanacirstea** [comment]
Most tests pass, but I see the `wasmedge-go-examples/go_Mobilenet` test fails with `segmentation violation`, when registering a Tensorflow module. 
I did not yet have time to run the test myself (tensorflow plugin does not yet work on macos arm), but at first glance, I could not see why the changes from this PR would be the cause of this failure.

Do you think this PR is the problem?

**Comment by q82419** [comment]
> Most tests pass, but I see the `wasmedge-go-examples/go_Mobilenet` test fails with `segmentation violation`, when registering a Tensorflow module. I did not yet have time to run the test myself (tensorflow plugin does not yet work on macos arm), but at first glance, I could not see why the changes from this PR would be the cause of this failure.
> 
> Do you think this PR is the problem?

Sorry that I'll check this in these days.

**Comment by q82419** [comment]
> Most tests pass, but I see the `wasmedge-go-examples/go_Mobilenet` test fails with `segmentation violation`, when registering a Tensorflow module. I did not yet have time to run the test myself (tensorflow plugin does not yet work on macos arm), but at first glance, I could not see why the changes from this PR would be the cause of this failure.
> 
> Do you think this PR is the problem?

This seems like the `ubuntu20.04` version of WasmEdge is not compatible with `manylinux2014_x86_64` version of `WasmEdge-TensorFlow` extensions.

Use the installer with `--dist manylinux2014` args will fix this.
For this issue, we'll discuss and make a better solution in the future.

**Review by q82419 — APPROVED** [review]

**Pull request merged**